peer/peer_test.go

```diff
@@ -481,6 +508,11 @@ func TestPeerListeners(t *testing.T) {
 		}
 	}
 
+	outPeer.QueueMessage(&UnknownMessage{}, nil)
```

Comment:
This should pass in a done channel so it can properly synchronize the assertion below. Can address this in a follow up change if the OP has abandoned the PR.

Reply:
I am not sure how to translate that suggestion to Go code :sob: